src/Markdig/Renderers/TextRendererBase.cs

```diff
@@ -27,8 +29,9 @@ public abstract class TextRendererBase : RendererBase
         /// <exception cref="ArgumentNullException"></exception>
         protected TextRendererBase(TextWriter writer)
         {
-            if (writer == null) ThrowHelper.ArgumentNullException_writer();
-            Writer = writer;
+            if (writer is null) ThrowHelper.ArgumentNullException_writer();
+            this.writer = writer;
+            this.writer.NewLine = "\n";
```

Comment:
```suggestion
            Writer = writer;
```
We don't gain anything by duplicating this

Reply:
Sadly, the analyzer doesn't follow the setter through and mark the writer field as non-null if we do this. We could set a MemberNotNull attribute and throw when setting the writer, but we also double up some logic if we do this.

Reply:
Unfortunate :/
This would be a fine place for `MemberNotNull`, but since we'd have to add a dummy class just for this, it's fine as is.